From 98df4af2ddc1922411d1d617280a87ec3c2ea140 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Mon, 26 Jan 2015 11:17:51 +0800
Subject: [PATCH 241/255] atomisp2: Fix copy pipe configuration for SOC sensor

Change-Id: Iefcd19784c83ff869809521cba079b703275e8ad
Fixes-Issue: #H1603928408
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/38436
Reviewed-by: Jian Xu Zheng <jian.xu.zheng@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c | 7 +++++--
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h   | 3 ++-
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 3b1000f..375e6f9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -4704,8 +4704,11 @@ static void atomisp_check_copy_mode(struct atomisp_sub_device *asd,
 	 * For SOC camera we are using copy_mode all the time
 	 * TDB: Do we have some cases where the copy_mode can't be used?
 	 */
-	if ((asd->isp->inputs[asd->input_curr].type == SOC_CAMERA) ||
-	    (!atomisp_subdev_format_conversion(asd, source_pad)))
+	if (((asd->isp->inputs[asd->input_curr].type == RAW_CAMERA) &&
+	    (!atomisp_subdev_format_conversion(asd, source_pad))) ||
+	    ((asd->isp->inputs[asd->input_curr].type == SOC_CAMERA) &&
+	    (asd->isp->inputs[asd->input_curr].camera_caps->
+		sensor[asd->sensor_curr].stream_num > 1)))
 		asd->copy_mode = true;
 	else
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
index 0221ed4..4c2453d 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
@@ -167,7 +167,8 @@
 	   sensor[asd->sensor_curr].stream_num == 1)
 
 #define ATOMISP_USE_YUVPP(asd)  \
-	(ATOMISP_SOC_CAMERA(asd) && ATOMISP_CSS_SUPPORT_YUVPP)
+	(ATOMISP_SOC_CAMERA(asd) && ATOMISP_CSS_SUPPORT_YUVPP && \
+	!asd->copy_mode)
 
 #define ATOMISP_DEPTH_SENSOR_STREAMON_COUNT 2
 
-- 
1.9.1

